### DISCUSSION TITLE
please explain buffers

### DISCUSSION THREAD

**Discussion by Hashed0719** [lavalink-devs/Lavalink#663] [Q&A] [answered]
i know nothing about java and audio systems can someone please explain what is `NAS buffer` and `frame buffer` in application.yml

**Comment by topi314** [comment, marked as answer]
the NAS buffer is a buffer for audio kept in native code where it isn't affected by java garbage collection pauses
this is done to reduce lags due to gc pauses 

the frame buffer is kept in the jvm and is simply how many ms of audio to keep in memory
